packages/web-components/src/select/select.ts

```diff
@@ -0,0 +1,206 @@
+
+  // Attributes
+  /**
+   * The name of the select element
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: name
+   */
+  @attr
+  public name?: string;
+
+  /**
+   * The required boolean attribute of the select element
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: required
+   */
+  @attr({ mode: 'boolean' })
+  public required?: boolean;
+
+  /**
+   * The optional label of the select
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: label
+   */
+  @attr
+  public label?: string;
+
+  /**
+   * The autofocus attribute of the select
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: autofocus
+   */
+  @attr({ attribute: 'autofocus', mode: 'boolean' })
+  public autofocusAttr?: boolean;
+
+  /**
+   * The autocomplete attribute of the select
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: autocomplete
+   */
+  @attr
+  public autocomplete?: string;
+
+  /**
+   * The size of the select
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: control-size
+   */
+  @attr
+  public controlSize?: SelectSize;
+
+  /**
+   * The disabled state of the select
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: disabled
+   */
+  @attr({ mode: 'boolean' })
+  public disabled?: boolean;
+
+  // Observables
+  /**
+   * The options of the select
+   *
+   * @public
+   */
+  @observable
+  public options?: HTMLOptionElement[];
+
+  /**
+   * A boolean value that triggers an input event
+   *
+   * @public
+   */
+  @observable
+  public inputEventTrigger: boolean = false;
+
+  /**
+   * A boolean value that triggers a change event
+   *
+   * @public
+   */
+  @observable
+  public changeEventTrigger: boolean = false;
+
+  // Events
+  private changeEvent: Event = new Event('change');
+  private inputEvent: Event = new Event('input');
+
+  // Getters and setters
+  // Returns the select element
+  public get selectElement(): HTMLSelectElement | null {
+    return this.shadowRoot?.querySelector('select') ?? null;
+  }
+
+  // Returns the index of the selected option
+  public get selectedIndex(): number {
+    return this.selectElement ? this.selectElement.selectedIndex : 0;
+  }
+
+  // Sets the index of the selected option
+  public set selectedIndex(value: number) {
+    if (this.selectElement) {
+      this.selectElement.selectedIndex = value;
+    }
+  }
+
+  // Returns the value of the selected option
+  public get value(): string {
+    return this.selectElement ? this.selectElement.value : '';
+  }
+
+  // Sets the value of the selected option
+  public set value(value: string) {
+    if (this.selectElement) {
+      this.selectElement.value = value;
+    }
+  }
+
+  // Returns the number of options in the select element
+  public get length(): number {
+    return this.selectElement ? this.selectElement.length : 0;
+  }
+
+  // Methods
+  public handleChange(): void {
+    this.dispatchEvent(this.changeEvent);
+  }
+
+  public handleInput(): void {
+    this.dispatchEvent(this.inputEvent);
+  }
```

Comment:
These should use the $emit helper -`this.$emit("change")` and `this.$emit("input")`

Reply:
This was for the (wrong) previous implementation that didn't use the emit helper.